src/charm.py

```diff
@@ -156,8 +164,10 @@ def _on_cluster_relation_changed(self, event: EventBase) -> None:
         # only restart where necessary to avoid slowdowns
         # config_changed call here implicitly updates jaas + zoo.cfg
         if (
-            self.config_manager.config_changed() or self.state.cluster.switching_encryption
-        ) and self.state.unit_server.started:
+            (self.config_manager.config_changed() or self.state.cluster.switching_encryption)
+            and self.state.unit_server.started
+            and self.upgrade_events.idle
+        ):
```

Comment:
It's now in the `events.upgrade.idle`, and we moved it down there.
For K8s, we do `idle` check differently, using `bool(self.upgrade_stack)` instead, so that logic is still there.

I moved it because we don't want to stop the updating of the quorum, just the restarts.